test/11_holograph_factory_tests.ts

```diff
@@ -1,61 +1,242 @@
-    it('should fail with invalid signature if signer is incorrect');
+  describe('deployHolographableContract()', async () => {
+    it('should fail with invalid signature if config is incorrect', async () => {
+      let { erc20Config, erc20ConfigHash, erc20ConfigHashBytes } = await generateErc20Config(
+        l1.network,
+        l1.deployer.address,
+        'hToken',
+        l1.network.tokenName + ' (Holographed #' + l1.network.holographId.toString() + ')',
+        'h' + l1.network.tokenSymbol,
+        l1.network.tokenName + ' (Holographed #' + l1.network.holographId.toString() + ')',
+        '1',
+        18,
+        ConfigureEvents([]),
+        generateInitCode(['address', 'uint16'], [l1.deployer.address, 0]),
+        l1.salt
+      );
+
+      let hTokenErc20Address = await l1.registry.getHolographedHashAddress(erc20ConfigHash);
+      hTokenErc20Address = await l1.registry.getHolographedHashAddress(erc20ConfigHash);
+
+      let sig = await l1.deployer.signMessage(erc20ConfigHashBytes);
+      let signature: Signature = StrictECDSA({
+        r: '0x' + sig.substring(2, 66),
+        s: '0x' + sig.substring(66, 130),
+        v: '0x' + sig.substring(130, 132),
+      } as Signature);
+
+      await expect(
+        l1.holographFactory.connect(deployer).deployHolographableContract(erc20Config, signature, owner.address)
+      ).to.be.revertedWith('HOLOGRAPH: invalid signature');
+    });
+
+    // TODO: Not sure how to test if each value of signature is correct
+    it.skip('should fail with invalid signature if signature.r is incorrect');
+    it.skip('should fail with invalid signature if signature.s is incorrect');
+    it.skip('should fail with invalid signature if signature.v is incorrect');
+
+    it('should fail with invalid signature if signer is incorrect', async () => {
+      let { erc20Config, erc20ConfigHash, erc20ConfigHashBytes } = await generateErc20Config(
+        l1.network,
+        l1.deployer.address,
+        'hToken',
+        l1.network.tokenName + ' (Holographed #' + l1.network.holographId.toString() + ')',
+        'h' + l1.network.tokenSymbol,
+        l1.network.tokenName + ' (Holographed #' + l1.network.holographId.toString() + ')',
+        '1',
+        18,
+        ConfigureEvents([]),
+        generateInitCode(['address', 'uint16'], [l1.deployer.address, 0]),
+        l1.salt
+      );
+
+      let hTokenErc20Address = await l1.registry.getHolographedHashAddress(erc20ConfigHash);
+      hTokenErc20Address = await l1.registry.getHolographedHashAddress(erc20ConfigHash);
+
+      let sig = await l1.deployer.signMessage(erc20ConfigHashBytes);
+      let signature: Signature = StrictECDSA({
+        r: '0x' + sig.substring(2, 66),
+        s: '0x' + sig.substring(66, 130),
+        v: '0x' + sig.substring(130, 132),
+      } as Signature);
+
+      await expect(l1.factory.deployHolographableContract(erc20Config, signature, nonOwner.address)).to.be.revertedWith(
+        'HOLOGRAPH: invalid signature'
```

Comment:
can you add 'HOLOGRAPH: invalid signature' to the `/test/utils/error_constants`. Or any other `revertWith` Functions.